cryodrgn/analysis.py

```diff
@@ -395,33 +395,42 @@ def plot_projections(imgs, labels=None):
             axes[i].set_title(labels[i])
     return fig, axes
 
-def gen_volumes(weights, config, zfile, outdir, cuda=None,
-                Apix=None, flip=False, downsample=None, invert=None):
+def gen_volumes(weights, config, zfile, outdir, device=None,
+                Apix=None, flip=False, downsample=None, invert=None, vol_start_index=0):
     '''Call cryodrgn eval_vol to generate volumes at specified z values
     Input:
         weights (str): Path to model weights .pkl
         config (str): Path to config.pkl
         zfile (str): Path to .txt file of z values
         outdir (str): Path to output directory for volumes,
-        cuda (int or None): Specify cuda device
+        device (int or None): Specify cuda device
         Apix (float or None): Apix of output volume
         flip (bool): Flag to flip chirality of output volumes
         downsample (int or None): Generate volumes at this box size
         invert (bool): Invert contrast of output volumes
+        vol_start_index (int): Start index for generated volumes
     '''
-    cmd = f'cryodrgn eval_vol {weights} --config {config} --zfile {zfile} -o {outdir}'
+    args = [
+        weights,
+        '--config', config,
+        '--zfile', zfile,
+        '-o', outdir
+    ]
     if Apix is not None:
-        cmd += f' --Apix {Apix}'
+        args += ['--Apix', f'{Apix}']
     if flip:
-        cmd += f' --flip'
+        args += ['--flip']
     if downsample is not None:
-        cmd += f' -d {downsample}'
+        args += ['-d', f'{downsample}']
     if invert:
-        cmd += f' --invert'
-    if cuda is not None:
-        cmd = f'CUDA_VISIBLE_DEVICES={cuda} {cmd}'
-    log(f'Running command:\n{cmd}')
-    return subprocess.check_call(cmd, shell=True)
```

Comment:
This invocation of `eval_vols` inside the `gen_volumes` function was happening through a subprocess instead of a simple function call. This makes it very difficult to invoke this code in the debugger, and is also not ideal since the spawned python process may not use the active python environment unless we're very careful. I've modified this to be a function call now.